### ISSUE TITLE
[Rust] code improperly highlighted

### ISSUE THREAD

**Issue by mouse07410** [sublimehq/Packages#2323]
Please see https://github.com/sharkdp/bat/issues/906 as this problem was discovered by another package using your highlighter.

- Sublime Version: <whatever Bat uses>
- OS Version: MacOS Catalina 10.15.4

``` rust
mod state {
    pub struct State<const S: &'static str> {
        total: u32,
    }

    impl State<"init"> {
        pub fn new() -> Self {
            Self { total: 0 }
        }

        pub fn accumulate(self) -> State<"accumulate"> {
            State { total: self.total }
        }
    }

    impl State<"accumulate"> {
        pub fn add(&mut self, add: u32) {
            self.total += add
        }

        pub fn freeze(self) -> State<"freeze"> {
            State { total: self.total }
        }
    }
}
```
Observer - 2nd `impl` (and everything that follows it) not highlighted properly.

![Screen Shot 2020-04-06 at 09 34 38](https://user-images.githubusercontent.com/5923577/78564024-da31bf00-77e9-11ea-8b67-028c007ca7d0.png)

**Comment by keith-hall** [comment]
there is a PR open which may fix this: https://github.com/sublimehq/Packages/pull/2305
(unfortunately I'm not in a position to check atm)

**Comment by mouse07410** [comment]
> unfortunately I'm not in a position to check atm

Since I'm using this package indirectly (`bat` uses it to highlight sources, and I install `bat` from its central repo), I'm not in a position to check it at all.

Hopefully, you'd be able to check it later?

**Comment by deathaxe** [comment]
This is what it looks like with the mensioned PR.

![grafik](https://user-images.githubusercontent.com/16542113/82120705-58d63080-9788-11ea-95be-8a21b74b5b47.png)

I'd say there's still an issue with the closing `>` in  `State<...>` but the overall look is a bit better.

**Comment by deathaxe** [comment]
Fixed by #2305
